chia/wallet/cat_wallet/cat_outer_puzzle.py

```diff
@@ -45,16 +44,18 @@ def get_inner_puzzle(self, constructor: PuzzleInfo, puzzle_reveal: Program) -> O
         if args is None:
             raise ValueError("This driver is not for the specified puzzle reveal")
         _, _, inner_puzzle = args
-        if constructor.also() is not None:
-            deep_inner_puzzle: Optional[Program] = self._get_inner_puzzle(constructor.also(), inner_puzzle)
+        also = constructor.also()
```

Comment:
the reason to name a variable for `also` is that mypy needs to know that we checked it against `None` in order to consider it a `PuzzleInfo` object. the method call `also()` always returns an `Optional[PuzzleInfo]` which isn't compatible with `_get_inner_puzzle()` (for example). It doesn't take an `Optional`